fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTestValidator.java

```diff
@@ -128,71 +126,102 @@ void validate(Index index, final Map<Tuple, Map<Tuple, Tuple>> expectedDocumentI
                     "group", groupingKey,
                     "expectedCount", entry.getValue().size()));
 
-            final Set<Map.Entry<Tuple, Tuple>> entries = entry.getValue().entrySet();
-            final List<Tuple> records = entries.stream()
-                    // I think in theory, this should be able to be:
-                    // Map.Entry.comparingByValue().thenComparing(Map.Entry.comparingByKey())
-                    // but I could not get the types to work
-                    .sorted((c1, c2) -> {
-                        final int valueComparison = c1.getValue().compareTo(c2.getValue());
-                        if (valueComparison != 0) {
-                            return valueComparison;
-                        } else {
-                            return c1.getKey().compareTo(c2.getKey());
-                        }
-                    })
+            final List<Tuple> records = entry.getValue().entrySet().stream()
+                    .sorted(Map.Entry.<Tuple, Tuple>comparingByValue().thenComparing(Map.Entry.comparingByKey()))
                     .map(Map.Entry::getKey)
                     .collect(Collectors.toList());
-            List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = getPartitionMeta(index, groupingKey);
-            partitionInfos.sort(Comparator.comparing(info -> Tuple.fromBytes(info.getFrom().toByteArray())));
-            final String allCounts = partitionInfos.stream()
-                    .map(info -> Tuple.fromBytes(info.getFrom().toByteArray()).toString() + info.getCount())
-                    .collect(Collectors.joining(",", "[", "]"));
-            Set<Integer> usedPartitionIds = new HashSet<>();
-            Tuple lastToTuple = null;
-            int visitedCount = 0;
+            if (partitionHighWatermark > 0) {
```

Comment:
Tip: This code is much easier to review if you hide whitespace changes, because the bulk of `validateUnpartitionedGroup` is the same, just indented.